### ISSUE TITLE
Create drasil-theory Package

### ISSUE THREAD

**Issue by samm82** [JacquesCarette/Drasil#918]
The `drasil-theory` subpackage should be created with the code for general requirements knowledge.

## TODO
- [x] Move GenDefn (#1320)
- [x] Move DataDefinition (#1338)
- [x] Move InstanceModel (#1331)
- [x] Move Theory (#1340)
- [x] Add to wiki page

_Each move should be its own PR_

**Comment by Mornix** [comment]
At least three of them (AssumpChunk, Change, ReqChunk) should be removed completely when #562 is done..

**Comment by szymczdm** [comment]
Exactly what @Mornix said. Also, that will likely require Labels to be implemented as well.

**Comment by samm82** [comment]
I've updated the README for the removed modules in #1242; should I proceed with moving the rest into docLang now that Label has been implemented? All prerequisite issues have been closed. @Mornix @szymczdm

**Comment by Mornix** [comment]
As far as I see, this should be “good-to-go” now, but I’d wait on a comment from @JacquesCarette to be certain.

**Comment by JacquesCarette** [comment]
Yes, please do go ahead.

**Comment by samm82** [comment]
Two questions @JacquesCarette @Mornix:

1. Should each individual move be its own PR? (ie. a PR for GenDefn, a PR for Theory, etc.)
2. What other chunks should be moved to docLang?
   - Citation
   - CommonIdea
   - Concept
   - DataDefinition
   - DefinedQuantity
   - Eq
   - GenDefn
   - InstanceModel
   - NamedIdea
   - Quantity
   - Relation'
   - Theory
   - UncertainQuantity
   - Unital
   - Unitary
   - UnitaryConcept
   - UnitDefn

**Comment by JacquesCarette** [comment]
Yes, each move should be its own PR.

`GenDefn`, `DataDefinition`, `InstanceModel` and `Theory` are all "requirements" knowledge. They are not, however, strictly "documentation" knowledge, they are more general than that. I think it would make sense to remove them from drasil-lang, but to put them into their own package. I am thinking drasil-theory.

**Comment by samm82** [comment]
`ChunkDB` uses these modules that are being moved to `drasil-theory`. Should it be moved there as well, or should it be split into parts, so there is a `ChunkDB` in `drasil-lang` and in `drasil-theory`? @JacquesCarette 

Also, my current folder structure of `drasil-theory` is Drasil > Theory > [modules], with a Theory.hs that re-exports everything in Drasil/. Let me know if you think it should be something different.

**Comment by JacquesCarette** [comment]
I think ChunkDB should be moved even further "out". I'm kind of thinking drasil-gen, but I'm not sure. We might need to create another package that takes care of the 'database' aspects of information collection.

**Comment by samm82** [comment]
For now, I'll move it to `drasil-gen` and see how that works out.

**Comment by Mornix** [comment]
You'll likely not be able to transplant it since `drasil-docLang` currently uses it.

**Comment by JacquesCarette** [comment]
Right, that was my concern.  So `drasil-database` it is!

**Comment by samm82** [comment]
Should the `Theory` module be renamed? It just seems a little weird to have `Theory.Drasil.Theory`. However, `Theory.hs` contains the definitions for `TheoryModel` and for the `Theory` class, so I guess it makes sense to have it named "Theory"?

**Comment by JacquesCarette** [comment]
A little weird, but until we have a better name, with a good reason, I'd leave it as is.
